Frontend js

billiamliu opened the pull request:
see commits for features.
KNOWN BUGS:
- recommendation is an empty array (need your help on server side)
- adding a recommended photos on my server result in 500 error, otherwise everything else is set up and needs testing (need your help on server side)
- search needs to be disabled for non-logged in users (i got this)

billiamliu committed:
added rec template, needs server merge to validate

billiamliu committed:
Merge branch 'staging' into frontendJS

merging from staging

billiamliu committed:
updated rec photo loading mechanism... wet code

billiamliu committed:
minor comment fixes

LaurelOlson committed:
added config for recommendation racoon

LaurelOlson committed:
Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into frontendJS

fixed conflict

billiamliu committed:
added theoretically working recommended photo styles, need handlers

billiamliu committed:
fixed merge conflict

billiamliu committed:
quick fix one hover to on hover

billiamliu committed:
quickfix to css

LaurelOlson committed:
fixed route for liked photos

LaurelOlson committed:
now actually fixed

LaurelOlson committed:
updated delete liked photo route

billiamliu committed:
added skeleton for recBtn handlers

billiamliu committed:
Merge branch 'frontendJS' of github.com:Jedeu/photon-ai into frontendJS

merging laurel's hotfix

LaurelOlson committed:
fixed liked photos route

billiamliu committed:
button frontend logic complete, missing server integration

billiamliu committed:
nav bar stats complete

billiamliu committed:
non users view top photos on page load

billiamliu committed:
search optimised, to do: disable login for non users

Jedeu commented:
Hey @billiamliu  great work and I feel like we're pretty much MVP-ready. A couple of things:
- Doesn't look like there's a way for me to go back to my view of photos after using fuzzy search without going into the side-bar and clicking "append real"
- Sometimes the "append real" photo doesn't end up appending all the photos.
- I can't append more photos once I nuke them. Any way to change that?

billiamliu committed:
made v2 of recPhoto ajax request in controller

LaurelOlson committed:
send response on post and delete req to likedphotos

billiamliu committed:
bugfix of only one rec photo showing

billiamliu committed:
added unlike button layout

LaurelOlson committed:
started implementing emotion tags

billiamliu committed:
unlike button works, tested, connects to all subsystems, and uses sessionStorage

LaurelOlson committed:
Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into frontendJS

'merge'

LaurelOlson committed:
removed billiam folder

billiamliu committed:
minor UI fix of rec and unlike button

LaurelOlson committed:
Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into emotion-detection

pull from frontendJS

LaurelOlson committed:
did nothing

billiamliu committed:
made minor change to rec button on success, should change to unlike

billiamliu committed:
persisted rec photo button

LaurelOlson committed:
merge frontendJS

Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into emotion-detection

billiamliu committed:
enabled emotions tags

LaurelOlson committed:
merge with frontendJS

Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into emotion-detection

LaurelOlson committed:
emotion detection now saves as a tag in the db

LaurelOlson committed:
merge to allow pull

Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into frontendJS

LaurelOlson committed:
removed emotion code - broken

billiamliu committed:
hotfix for emotions model typo

LaurelOlson committed:
Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into frontendJS

LaurelOlson committed:
merge

billiamliu committed:
search feature added emotions, btn colour fix

LaurelOlson committed:
fixed conflict

LaurelOlson committed:
merge

Merge branch 'frontendJS' of https://github.com/Jedeu/photon-ai into frontendJS

billiamliu committed:
rec photos can now flip over to liked photos

billiamliu committed:
Merge branch 'frontendJS' of github.com:Jedeu/photon-ai into frontendJS

merging laurel's changes

billiamliu commented:
fixed and will pull again
